crates/router/src/connector/trustpay/transformers.rs

```diff
@@ -435,24 +435,30 @@ fn get_pending_status_based_on_redirect_url(redirect_url: Option<Url>) -> enums:
 }
 
 fn get_transaction_status(
-    payment_status: &str,
+    payment_status: Option<String>,
     redirect_url: Option<Url>,
 ) -> CustomResult<(enums::AttemptStatus, Option<String>), errors::ConnectorError> {
-    let (is_failed, failure_message) = is_payment_failed(payment_status);
-    let pending_status = get_pending_status_based_on_redirect_url(redirect_url);
-    if payment_status == "000.200.000" {
-        return Ok((pending_status, None));
-    }
-    if is_failed {
-        return Ok((
-            enums::AttemptStatus::AuthorizationFailed,
-            Some(failure_message.to_string()),
-        ));
-    }
-    if is_payment_successful(payment_status)? {
-        return Ok((enums::AttemptStatus::Charged, None));
+    // We don't get payment_status only in case, when the user doesn't complete the authentication step.
+    // If we receive status, then return the proper status based on the connector response
+    if let Some(payment_status) = payment_status {
+        let (is_failed, failure_message) = is_payment_failed(&payment_status);
+        if is_failed {
+            return Ok((
+                enums::AttemptStatus::AuthorizationFailed,
+                Some(failure_message.to_string()),
+            ));
+        }
+
+        if is_payment_successful(&payment_status)? {
+            return Ok((enums::AttemptStatus::Charged, None));
+        }
+
+        let pending_status = get_pending_status_based_on_redirect_url(redirect_url);
```

Comment:
@sai-harsha-vardhan If redirect_url is None we are setting payment status as Authorizing rather than Pending ?

Reply:
@Sangamesh26 get_pending_status_based_on_redirect_url should return pending incase redirect_url is None